File: src/main/java/com/cse3063f25grp1/reranker/Reranker.java
package com.cse3063f25grp1.reranker;

import java.util.List;
import com.cse3063f25grp1.data.ChunkStore;
import com.cse3063f25grp1.model.Hit;

public interface Reranker {
    List<Hit> rerank(List<String> query, List<Hit> hits, ChunkStore store);
}

File: iteration2/src/reranker/reranker.py
from abc import ABC, abstractmethod
from typing import List

from src.data.chunk_store import ChunkStore
from src.model.hit import Hit

class Reranker(ABC):

    @abstractmethod
    def rerank(self, query: List[str], hits: List[Hit], store: ChunkStore) -> List[Hit]:
        pass
